File: java/src/phase 1-conditional_thinking_java/Q31CharClassification.java
import java.util.Scanner;

/**
 * Phase 1: Conditional Thinking (If-Else, Boolean Logic)
 * Level 4: Logical Operators & Compound Statements
 * Question 31: Take a character and check if it is a letter, a digit, or neither.
 * 
 * Mentor Tips (Java 26):
 * - Python: char.isalpha() or char.isdigit(). Java: Character.isLetter() or Character.isDigit().
 */
public class Q31CharClassification {

    public static String solve(char charVal) {
        // TODO: Implement this method to solve the question
        return "";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Char: "); char c = scanner.next().charAt(0);\n        System.out.println("Classification: " + solve(c));
        scanner.close();
    }
}

File: python/phase 1-condtional_thinking/q31_char_classification.py
"""
Phase 1: Conditional Thinking (If-Else, Boolean Logic)
Level 4: Logical Operators & Compound Statements
Question 31: Take a character and check if it is a letter, a digit, or neither.

Mentor Tips (Python 3.14+):
- Python: char.isalpha() or char.isdigit(). Java: Character.isLetter() or Character.isDigit().
"""

def solve(char: str) -> str:
    # TODO: Implement this method to solve the question
    if char.isalpha():
        return "letter"
    elif char.isdigit():
        return "digit"
    else:
        return "neither"

def main():
    char = input("Char: ")
    print(f"Classification: {solve(char)}")

if __name__ == "__main__":
    main()
